spec/requests/profile_spec.rb

```diff
@@ -0,0 +1,166 @@
+# frozen_string_literal: true
+
+describe "Profile", type: :request do
+  describe "#index" do
+    context "when user is not logged in" do
+      before { get profile_index_path, params: {uid: "user-id"} }
+      it { expect(response).to redirect_to(new_user_session_path) }
+    end
+
+    context "when user is logged in" do
+      let(:user) { create(:user1) }
+      before { sign_in user }
+      it "responds successfully" do
+        get profile_index_path, params: {uid: user.uid}
+        expect(response).to be_successful
+      end
+
+      context "when profile belongs to logged in user" do
+        it "returns that users profile" do
+          get profile_index_path, params: {uid: user.uid}
+          expect(response.body).to include(user.name)
+        end
+
+        context "when user has no moments and strategies" do
+          it "does not have the Stories section" do
+            get profile_index_path, params: {uid: user.uid}
+            expect(response.body).to_not include("Stories")
+          end
+        end
+
+        context "when user has moments or strategies" do
+          it "has the Stories section" do
+            moment = create(:moment, user: user)
+            strategy = create(:strategy, user: user)
+
+            get profile_index_path, params: {uid: user.uid}
+            expect(response.body).to include("Stories")
+            expect(response.body).to include(moment.name)
+            expect(response.body).to include(strategy.name)
+          end
+        end
+      end
+
+      context "when profile does not belong to user" do
+        let(:profile_owner) { build_stubbed(:user2) }
+
+        before do
+          allow(User).to receive(:find_by).and_return(profile_owner)
+          get profile_index_path, params: {uid: profile_owner.uid}
+        end
+
+        it "does not have the Stories section" do
+          expect(response.body).to_not include("Stories")
+        end
+
+        it "returns the profile for the owner" do
+          expect(response.body).to include(profile_owner.name)
+        end
+      end
+    end
+  end
+
+  describe "#data" do
+    let(:user) { create(:user) }
+    let!(:moment) { create(:moment, user: user) }
+
+    context "when the user is signed in" do
```

Comment:
if user is signed in context exists, there should be a user is not signed in context also.

Test all use cases

Reply:
Line 76 is the user not signed in context